src/main/java/com/fundynamic/d2tm/game/map/MapRenderer.java

```diff
@@ -29,8 +33,35 @@ private void renderMap(Graphics graphics, Map map) throws SlickException {
         for (int x = 1; x <= map.getWidth(); x++) {
             for (int y = 1; y <= map.getHeight(); y++) {
                 Cell cell = map.getCell(x, y);
-                graphics.drawImage(cell.getTileImage(), (x - 1) * tileWidth, (y - 1) * tileHeight);
+                int cellX = (x - 1) * tileWidth;
+                int cellY = (y - 1) * tileHeight;
+                graphics.drawImage(cell.getTileImage(), cellX, cellY);
+
+                ShroudFacing shroudFacing = determineFacing(map, x, y);
+                if (shroudFacing != null) {
```

Comment:
We once discussed adding another 'invisible' row of cells on each side of the map. So when you have a 512x512 map, it's actually 514x514. That way there is always a cell to spawn reinforcement from, and we can skip the bounds-check for these kinds of algorithms.